### PR TITLE
add apeswap banana-matic pool

### PR DESCRIPTION
**Pull request by Crypto-One-dev**
Add Apeswap MiniApev2 strategy for `banana-matic` pair

image

We will receive `banana` as reward from the Ape pool.

> Polygon Mainnet ApeSwap BANANA/MATIC
> Impersonating...
> 0xf00dD244228F51547f0563e60bCa65a30FBF5f7f
> Fetching Underlying at:  0x034293F21F1cCE5908BC605CE5850dF2b1059aC0
> New Vault Deployed:  0x885D59830C1FdB120B54d62790dB7A6a1f534463
> Strategy Deployed:  0x1caD479Dd2e9e86457594E4cEA3D4bdfE67fE373
> Strategy and vault added to Controller.
> farmerBalance:
> 2473386633013588677856
>     Happy path
> loop  0
> old shareprice:  1000000000000000000
> new shareprice:  1000000000000000000
> growth:  1
> loop  1
> old shareprice:  1000000000000000000
> new shareprice:  1000349469853992441
> growth:  1.0003494698539925
> loop  2
> old shareprice:  1000349469853992441
> new shareprice:  1000699410051572170
> growth:  1.000349817946753
> loop  3
> old shareprice:  1000699410051572170
> new shareprice:  1001049377899076040
> growth:  1.0003497232475493
> loop  4
> old shareprice:  1001049377899076040
> new shareprice:  1001399454201590828
> growth:  1.0003497093252776
> loop  5
> old shareprice:  1001399454201590828
> new shareprice:  1001749666160654332
> growth:  1.0003497225383877
> loop  6
> old shareprice:  1001749666160654332
> new shareprice:  1002099986749344665
> growth:  1.0003497087151854
> loop  7
> old shareprice:  1002099986749344665
> new shareprice:  1002450429529935037
> growth:  1.0003497083975892
> loop  8
> old shareprice:  1002450429529935037
> new shareprice:  1002800994557138840
> growth:  1.000349708092168
> loop  9
> old shareprice:  1002800994557138840
> new shareprice:  1003151695442003610
> growth:  1.0003497213173584
> earned!
> APR: 61.364127280912825 %
> APY: 84.61936270154216 %
>       ✔ Farmer should earn money (182358ms)

### CODE REVIEW

File: contracts/base/ape-base/MiniApeV2Strategy.sol [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+        uniLPComponentToken1,
+        token0Amount,
+        token1Amount,
+        1,  // we are willing to take whatever the pair gives us
+        1,  // we are willing to take whatever the pair gives us
+        address(this),
+        block.timestamp
+      );
+    } else {
+      IUniswapV2Router02(apeswapRouterV2).swapExactTokensForTokens(
```
Comment (belbix) [inline contracts/base/ape-base/MiniApeV2Strategy.sol:242]: make a test for single token or remove this logic

File: contracts/base/ape-base/MiniApeV2Strategy.sol [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+  /**
+  * Can completely disable claiming UNI rewards and selling. Good for emergency withdraw in the
+  * simplest possible way.
+  */
+  function setSell(bool s) public onlyGovernance {
+    _setSell(s);
+  }
+
+  /**
+  * Sets the minimum amount of CRV needed to trigger a sale.
```
Comment (belbix) [inline contracts/base/ape-base/MiniApeV2Strategy.sol:341]: fix comment

File: contracts/base/ape-base/MiniApeV2Strategy.sol [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+  *   when the investing is being paused by governance.
+  */
+  function doHardWork() external onlyNotPausedInvesting restricted {
+    harvestReward();
+    _liquidateReward();
+    investAllUnderlying();
+  }
+
+  /**
+  * Can completely disable claiming UNI rewards and selling. Good for emergency withdraw in the
```
Comment (belbix) [inline contracts/base/ape-base/MiniApeV2Strategy.sol:333]: fix comment

File: contracts/base/ape-base/MiniApeV2Strategy.sol [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    uniswapRoutes[_token] = _route;
+  }
+
+  // We assume that all the tradings can be done on Uniswap
+  function _liquidateReward() internal {
+    // we can accept 1 as minimum because this is called only by a trusted role
+    uint256 amountOutMin = 1;
+
+    // swap second reward token to reward token
+    uint256 secondRewardBalance = IERC20(secondRewardToken()).balanceOf(address(this));
```
Comment (belbix) [inline contracts/base/ape-base/MiniApeV2Strategy.sol:148]: I think we should have sellFloor for the second reward too
please add a new variable and setter

File: contracts/base/ape-base/interface/IRewarder.sol [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+import "../lib/BoringERC20.sol";
+interface IRewarder {
+    using BoringERC20 for IERC20;
```
Comment (belbix) [inline contracts/base/ape-base/interface/IRewarder.sol:6]: remove

File: contracts/base/ape-base/interface/IRewarder.sol [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+import "../lib/BoringERC20.sol";
```
Comment (belbix) [inline contracts/base/ape-base/interface/IRewarder.sol:4]: remove

File: contracts/base/ape-base/lib/BoringERC20.sol [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.6.12;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+library BoringERC20 {
```
Comment (belbix) [inline contracts/base/ape-base/lib/BoringERC20.sol:5]: remove, we don't need this lib

**Review by belbix — CHANGES_REQUESTED** [review]

**Review by belbix — APPROVED** [review]

**Review by CryptJS13 — APPROVED** [review]
@Cryo45 @breadforthepeople Had a look at Apeswap too. It's a perfect Sushi clone. Only difference is removal of "migrator" code, which is a good thing.

It originated on BSC together with all the other Sushi clones, but somehow it survived till now. Could make decent vaults, its worth a try in my opinion.

Deployed these 7 vaults, @Crypto-One-dev will add them to the API.

**Pull request merged**